src/NzbDrone.Core/Notifications/CustomScript/CustomScriptService.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Collections.Specialized;
+using System.IO;
+using System.Linq;
+using FluentValidation.Results;
+using NLog;
+using NzbDrone.Common.Processes;
+using NzbDrone.Core.MediaFiles;
+using NzbDrone.Core.Tv;
+
+namespace NzbDrone.Core.Notifications.CustomScript
+{
+    public interface ICustomScriptService
+    {
+        void OnDownload(Series series, EpisodeFile episodeFile, CustomScriptSettings settings);
+        void AfterRename(Series series, CustomScriptSettings settings);
+        ValidationFailure Test(CustomScriptSettings settings);
+    }
+
+    public class CustomScriptService : ICustomScriptService
+    {
+        private readonly IProcessProvider _processProvider;
+        private readonly Logger _logger;
+
+        public CustomScriptService(IProcessProvider processProvider, Logger logger)
+        {
+            _processProvider = processProvider;
+            _logger = logger;
+        }
+
+        public void OnDownload(Series series, EpisodeFile episodeFile, CustomScriptSettings settings)
+        {
+            var environmentVariables = new StringDictionary();
+
+            environmentVariables.Add("Sonarr.EventType", "Download");
```

Comment:
so, this won't allow you to run a 3rd party program directly? you would always have to write a wrapping script that uses the evn variables and passes them as args?